src/SimpleList/index.js

```diff
@@ -54,9 +54,15 @@ export default class SimpleList extends Component {
   filterItems(items) {
     let { currentQuery } = this.state
     return items.filter((itm) => {
-      if (itm.firstLine.text.indexOf(currentQuery) >= 0) {
+      if (!currentQuery) {
         return true
-      } else if (itm.secondLine.text.indexOf(currentQuery) >= 0) {
+      }
+      if (itm.firstLine.text && itm.firstLine.text.indexOf(currentQuery) >= 0) {
+        return true
+      } else if (
+        itm.secondLine.text &&
+        itm.secondLine.text.indexOf(currentQuery) >= 0
+      ) {
```

Comment:
These two `filterItems` functions are so close to being identical that it might be worth extracting them out into one util and writing tests for them:

```
  filterItems(items, title = 'title', subtitle = 'subtitle') {
    const { currentQuery } = this.state
    return items.filter(
      itm =>
        !currentQuery ||
        (itm[title].text && itm[title].text.indexOf(currentQuery) >= 0) ||
        (itm[subtitle].text && itm[subtitle].text.indexOf(currentQuery) >= 0)
    )
  }
```
And then you'd call it like `filterItems(items, 'firstLine', 'secondLine')` from here and `filterItems(items)` from the `<ImageList>` component.

I refactored it ☝️ in a way that is most readable for me, it doesn't particularly matter if you do it that way or leave the `if` checks, but we should probably write some tests for them, and it would be easier to do so if they were consolidated down to one function.

Reply:
Turns out we're missing the scaffolding required to write tests in this repo, and implementing it is out-of-scope.